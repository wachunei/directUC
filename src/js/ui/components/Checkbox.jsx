import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const Label = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;

  ${(props) =>
    props.disabled &&
    css`
      cursor: not-allowed;
      filter: grayscale(100%) opacity(70%);
    `};
`;

const CheckMark = styled.span`
  display: inline-block;
  width: 1rem;
  height: 1rem;
  background: ${(props) => props.theme.checkboxBackground};
  border: 1px solid ${(props) => props.theme.inputBorder};
  margin-right: 0.8rem;
  flex: none;
  position: relative;

  &::after {
    display: block;
    opacity: 0;
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 0.4rem;
    height: 0.8rem;
    border: 2px solid ${(props) => props.theme.checkMark};
    border-width: 0 2px 2px 0;
    transform: translate(100%, 15%) rotate(45deg);
    transform-origin: 100% 0%;
    transition: opacity 0.05s;
  }

  ${Label}:hover & {
    border: 1px solid ${(props) => props.theme.checkboxHoverBorder};
  }
`;

const CheckboxInput = styled.input.attrs({
  type: "checkbox",
})`
  position: absolute;
  opacity: 0;
  pointer-events: none;
  &:focus ~ ${CheckMark} {
    border: 1px solid ${(props) => props.theme.checkboxFocusBorder};
    outline: 1px solid ${(props) => props.theme.checkboxFocusBorder};
  }
  &:checked ~ ${CheckMark}::after {
    opacity: 1;
  }
`;

function Checkbox({ children, disabled, ...props }) {
  return (
    <Label disabled={disabled}>
      <CheckboxInput {...props} disabled={disabled} />
      <CheckMark />
      <span>{children}</span>
    </Label>
  );
}

Checkbox.propTypes = {
  children: PropTypes.node,
  disabled: PropTypes.bool,
};

Checkbox.defaultProps = {
  children: "",
  disabled: false,
};

export default Checkbox;
