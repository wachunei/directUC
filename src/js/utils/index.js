export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
