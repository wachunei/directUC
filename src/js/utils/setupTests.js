// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
