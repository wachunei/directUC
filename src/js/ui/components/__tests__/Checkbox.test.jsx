import React from "react";
import Checkbox from "../Checkbox";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Checkbox Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Checkbox>A test checkbox</Checkbox>);
    expect(tree).toMatchSnapshot();
  });

  test("should render checked", () => {
    const tree = renderTree(<Checkbox checked>A test checkbox</Checkbox>);
    expect(tree).toMatchSnapshot();
  });
  test("should render disabled", () => {
    const tree = renderTree(<Checkbox disabled>A test checkbox</Checkbox>);
    expect(tree).toMatchSnapshot();
  });
});
