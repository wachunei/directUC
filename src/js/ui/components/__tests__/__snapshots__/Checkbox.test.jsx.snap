// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c4 {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  background: #FFFFFF;
  border: 1px solid #909090;
  margin-right: 0.8rem;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  position: relative;
}

.c4::after {
  display: block;
  opacity: 0;
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0.4rem;
  height: 0.8rem;
  border: 2px solid #1F95EB;
  border-width: 0 2px 2px 0;
  -webkit-transform: translate(100%,15%) rotate(45deg);
  -ms-transform: translate(100%,15%) rotate(45deg);
  transform: translate(100%,15%) rotate(45deg);
  -webkit-transform-origin: 100% 0%;
  -ms-transform-origin: 100% 0%;
  transform-origin: 100% 0%;
  -webkit-transition: opacity 0.05s;
  transition: opacity 0.05s;
}

.c0:hover .c4 {
  border: 1px solid #1F95EB;
}

.c2 {
  position: absolute;
  opacity: 0;
  pointer-events: none;
}

.c2:focus ~ .c3 {
  border: 1px solid #1F95EB;
  outline: 1px solid #1F95EB;
}

.c2:checked ~ .c3::after {
  opacity: 1;
}

<span
  className="c0 c1"
  disabled={false}
>
  <input
    className="c2"
    disabled={false}
    type="checkbox"
  />
  <span
    className="c3 c4"
  />
  <span>
    A test checkbox
  </span>
</span>
`;

exports[`Checkbox Component should render checked 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c4 {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  background: #FFFFFF;
  border: 1px solid #909090;
  margin-right: 0.8rem;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  position: relative;
}

.c4::after {
  display: block;
  opacity: 0;
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0.4rem;
  height: 0.8rem;
  border: 2px solid #1F95EB;
  border-width: 0 2px 2px 0;
  -webkit-transform: translate(100%,15%) rotate(45deg);
  -ms-transform: translate(100%,15%) rotate(45deg);
  transform: translate(100%,15%) rotate(45deg);
  -webkit-transform-origin: 100% 0%;
  -ms-transform-origin: 100% 0%;
  transform-origin: 100% 0%;
  -webkit-transition: opacity 0.05s;
  transition: opacity 0.05s;
}

.c0:hover .c4 {
  border: 1px solid #1F95EB;
}

.c2 {
  position: absolute;
  opacity: 0;
  pointer-events: none;
}

.c2:focus ~ .c3 {
  border: 1px solid #1F95EB;
  outline: 1px solid #1F95EB;
}

.c2:checked ~ .c3::after {
  opacity: 1;
}

<span
  className="c0 c1"
  disabled={false}
>
  <input
    checked={true}
    className="c2"
    disabled={false}
    type="checkbox"
  />
  <span
    className="c3 c4"
  />
  <span>
    A test checkbox
  </span>
</span>
`;

exports[`Checkbox Component should render disabled 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(70%);
  filter: grayscale(100%) opacity(70%);
}

.c4 {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  background: #FFFFFF;
  border: 1px solid #909090;
  margin-right: 0.8rem;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  position: relative;
}

.c4::after {
  display: block;
  opacity: 0;
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0.4rem;
  height: 0.8rem;
  border: 2px solid #1F95EB;
  border-width: 0 2px 2px 0;
  -webkit-transform: translate(100%,15%) rotate(45deg);
  -ms-transform: translate(100%,15%) rotate(45deg);
  transform: translate(100%,15%) rotate(45deg);
  -webkit-transform-origin: 100% 0%;
  -ms-transform-origin: 100% 0%;
  transform-origin: 100% 0%;
  -webkit-transition: opacity 0.05s;
  transition: opacity 0.05s;
}

.c0:hover .c4 {
  border: 1px solid #1F95EB;
}

.c2 {
  position: absolute;
  opacity: 0;
  pointer-events: none;
}

.c2:focus ~ .c3 {
  border: 1px solid #1F95EB;
  outline: 1px solid #1F95EB;
}

.c2:checked ~ .c3::after {
  opacity: 1;
}

<span
  className="c0 c1"
  disabled={true}
>
  <input
    className="c2"
    disabled={true}
    type="checkbox"
  />
  <span
    className="c3 c4"
  />
  <span>
    A test checkbox
  </span>
</span>
`;
